Colab-TextGen-GPU.ipynb

```diff
@@ -0,0 +1,144 @@
+    },
+    "language_info": {
+      "name": "python"
+    },
+    "accelerator": "GPU"
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "view-in-github",
+        "colab_type": "text"
+      },
+      "source": [
+        "<a href=\"https://colab.research.google.com/github/Troys-Code/AI-Automation/blob/main/Colab-TextGen-GPU.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# oobabooga/text-generation-webui\n",
+        "\n",
+        "After running both cells, a public gradio URL will appear at the bottom in a few minutes. You can optionally generate an API link.\n",
+        "\n",
+        "* Project page: https://github.com/oobabooga/text-generation-webui\n",
+        "* Gradio server status: https://status.gradio.app/"
+      ],
+      "metadata": {
+        "id": "MFQl6-FjSYtY"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#@title 1. Keep this tab alive to prevent Colab from disconnecting you { display-mode: \"form\" }\n",
+        "\n",
+        "#@markdown Press play on the music player that will appear below:\n",
+        "%%html\n",
+        "<audio src=\"https://oobabooga.github.io/silence.m4a\" controls>"
+      ],
+      "metadata": {
+        "id": "f7TVVj_z4flw"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#@title 2. Launch the web UI\n",
+        "\n",
+        "#@markdown If unsure about the branch, write \"main\" or leave it blank.\n",
+        "\n",
+        "import torch\n",
+        "from pathlib import Path\n",
+        "\n",
+        "if Path.cwd().name != 'text-generation-webui':\n",
+        "  print(\"Installing the webui...\")\n",
+        "\n",
+        "  !git clone https://github.com/oobabooga/text-generation-webui\n",
+        "  %cd text-generation-webui\n",
+        "\n",
+        "  torver = torch.__version__\n",
+        "  print(f\"TORCH: {torver}\")\n",
+        "  is_cuda118 = '+cu118' in torver  # 2.1.0+cu118\n",
+        "  is_cuda117 = '+cu117' in torver  # 2.0.1+cu117\n",
+        "\n",
+        "  textgen_requirements = open('requirements.txt').read().splitlines()\n",
+        "  if is_cuda117:\n",
+        "      textgen_requirements = [req.replace('+cu121', '+cu117').replace('+cu122', '+cu117').replace('torch2.1', 'torch2.0') for req in textgen_requirements]\n",
+        "  elif is_cuda118:\n",
+        "      textgen_requirements = [req.replace('+cu121', '+cu118').replace('+cu122', '+cu118') for req in textgen_requirements]\n",
+        "  with open('temp_requirements.txt', 'w') as file:\n",
+        "      file.write('\\n'.join(textgen_requirements))\n",
+        "\n",
+        "  !pip install -r extensions/openai/requirements.txt --upgrade\n",
```

Comment:
**Security Issue**: The code snippet for cloning and installing dependencies directly in the notebook presents a security risk. Running shell commands that install packages or clone repositories without verifying the integrity or authenticity of the source can lead to the execution of malicious code. <br> **Fix**: Ensure that any git repository or package is verified against a known good hash or signature before installation. Additionally, consider pinning dependencies to specific versions to avoid unexpected changes. <br> **Code Suggestion**: 
 ```
 if Path.cwd().name != 'text-generation-webui':
   print("Verifying repository integrity...")
   # Calculate the hash of the cloned repository
   repo_hash = hashlib.sha256()
   with open('text-generation-webui', 'rb') as f:
     while True:
         data = f.read(repo_hash.block_size)
         if not data:
             break
         repo_hash.update(data)
   # Compare the calculated hash with the known good hash
   known_good_hash = 'EXPECTED_HASH_VALUE'
   if repo_hash.hexdigest() == known_good_hash:
     print("Repository integrity verified.")
   else:
     raise ValueError("Repository integrity verification failed.")
 ```